packages/flutter/lib/src/material/radio.dart

```diff
@@ -303,18 +304,43 @@ class Radio<T> extends StatefulWidget {
 
   /// The color for the radio's [Material] when it has the input focus.
   ///
+  /// If [overlayColor] returns a non-null color in the [MaterialState.focused]
```

Comment:
same for all these comments. "If null, then the value of [overlayColor] in the <> state is used... If that is null, [RadioThemeData.overlayColor] is used. If that is also null, etc."

Reply:
That is not exactly what is happening here, [overlayColor] is overriding the value of `focusColor` and `hoverColor`. I wanted to make it extra clear, that is why it is on its own row.

I am basing it on how it was done for activeColor, see docs: https://master-api.flutter.dev/flutter/material/Checkbox/activeColor.html

Do you still think I should combine it?

Reply:
Ah okay, I understand. Yes, it was hard to understand due to the non-specific 'it' and 'this'. How about removing 'of this color'?

Reply:
That's a good idea. Changed to that now!